numpy/core/src/npysort/quicksort.c.src

```diff
@@ -198,6 +208,7 @@ stack_pop:
         }
         pr = *(--sptr);
         pl = *(--sptr);
+        cdepth = *(--psdepth);
```

Comment:
but this switches everything to heapsort on the first bad partition, a bad partition does not mean the full array needs to be heapsorted only the current part